### PR TITLE
question: bug fix for issue no warning when editing a comment to an empty comment #3411

### LINKED ISSUE #3411: [bug] No warning when editing a comment to an empty comment
**Describe the bug**
There is no warning or error message that notifies the user that an edited comment cannot be empty.

**To Reproduce**
Steps to reproduce the behaviour:

1. Write a comment eg. on one of the research threads
2. Click on Edit comment
3. Remove all the characters and make an empty comment then save
4. It doesn't change the original comment

**Expected behaviour**
When the save button is clicked, display a warning message that a comment cannot be empty

**Screenshots**
Screenshot 2024-04-02 at 11 44 59 PM
Screenshot 2024-04-02 at 11 45 07 PM
Screenshot 2024-04-02 at 11 45 23 PM

**Additional context**
I am a newbie here and want to try fixing this bug myself first if that is okay!

### PR DESCRIPTION
**Pull request by npaddle**
PR Checklist

- [x ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

made edits to QuestionDiscussion.tsx to now give users a warning when trying to submit a blank edited comment (with blank space). Accomplished this by adding more validation checks as well as an error message to alert users on this specification.

## Git Issues

Closes #3411

## Screenshots/Videos

Screenshot 2024-04-26 112047
Screenshot 2024-04-26 112030

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by benfurber** [comment]
@npaddle Thanks so much for your first time contribution! 

If you have a read of my comments on #3411 as I think a different approach should be used rather than having `src/pages/Question/QuestionDiscussion.tsx` to have to worry about it. I think changes to the `EditComment` component (`packages/component/src/EditComment`) would be better as it would fix this issue for the other content types that also have this problem.

Did you have an ideas on the specific implementation we could go with @V24039?

**Comment by V24039** [comment]
@benfurber Instead of making changes in `src/pages/Question/QuestionDiscussion.tsx` we can validate at `packages/component/src/EditComment` field component and create a function for the save button which will check the value and then call the props.handleSubmit
Something like this 
image

This will also add the error style used in the Field component so no new styling will be required. 
Will need an error message to show to the user.

**Comment by benfurber** [comment]
@npaddle Are you able to give it another go or should @V24039 give it a try?

**Comment by benfurber** [comment]
@V24039 - As we haven't heard from @npaddle, are you still up for giving this a go please?

**Comment by V24039** [comment]
I will raise a PR ASAP

File: src/pages/Question/QuestionDiscussion.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ export const QuestionDiscussion = (props: IProps) => {
   const [comment, setComment] = useState('')
   const [discussion, setDiscussion] = useState<IDiscussion | null>(null)
   const [isLoading, setIsLoading] = useState(true)
+  const [errorMessage, setErrorMessage] = useState('') // State to track error messages
```
Comment (codisart) [inline src/pages/Question/QuestionDiscussion.tsx:20]: quibble: It seems that the comment is redundant with the name of the state.

File: src/pages/Question/QuestionDiscussion.tsx [hunk 2]
```diff
@@ -47,15 +48,24 @@ export const QuestionDiscussion = (props: IProps) => {
   }, [questionDocId])
 
   const handleEdit = async (_id: string, comment: string) => {
-    if (!discussion) return
+    if (comment.trim().length == 0 || comment.length == 0) {
```
Comment (codisart) [inline src/pages/Question/QuestionDiscussion.tsx:51]: suggestion: If the comment is already of lenght 0 then comment.trim() would be also 0. It seems that the second part of the condition is unnecessary.

File: src/pages/Question/QuestionDiscussion.tsx [hunk 3]
```diff
@@ -67,6 +77,11 @@ export const QuestionDiscussion = (props: IProps) => {
   }
 
   const onSubmit = async (comment: string) => {
+    if (comment.trim() === '' || comment.length == 0) {
+      setErrorMessage('Comment cannot be blank')
```
Comment (codisart) [inline src/pages/Question/QuestionDiscussion.tsx:81]: suggestion: use the same message everywhere l.52 l.81 l.99

**Comment by benfurber** [comment]
Thanks for the comments @codisart, but I'm going to close this now as @V24039 has a PR this: #3502

**Pull request closed without merging**